test/cbmc/stubs/core_sntp_stubs.c

```diff
@@ -0,0 +1,183 @@
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_sntp_stubs.c
+ * @brief Stubs to generate client authentication codes and validate server authentication.
+ */
+
+#include <stdint.h>
+#include "core_sntp_client.h"
+#include "core_sntp_serializer.h"
+#include "core_sntp_stubs.h"
+
+#define TEST_TIMESTAMP         \
+    {                          \
+        .seconds = UINT32_MAX, \
+        .fractions = 1000      \
+    }
+
+/* An exclusive bound on the times that the NetworkInterfaceSendStub will be
+ * invoked before returning a loop terminating value. This is usually defined
+ * in the Makefile of the harnessed function. */
+#ifndef MAX_NETWORK_SEND_TRIES
+    #define MAX_NETWORK_SEND_TRIES    3
+#endif
+
+/* An exclusive bound on the times that the NetworkInterfaceReceiveStub will
+ * return an unbound value. At this value and beyond, the
+ * NetworkInterfaceReceiveStub will return zero on every call. */
+#ifndef MAX_NETWORK_RECV_TRIES
+    #define MAX_NETWORK_RECV_TRIES    4
+#endif
+
+static SntpTimestamp_t testTime = TEST_TIMESTAMP;
+
+int32_t NetworkInterfaceReceiveStub( NetworkContext_t * pNetworkContext,
+                                     uint32_t serverAddr,
+                                     uint16_t serverPort,
+                                     void * pBuffer,
+                                     size_t bytesToRecv )
+{
+    __CPROVER_assert( pBuffer != NULL,
+                      "NetworkInterfaceReceiveStub pBuffer is NULL." );
+
+    __CPROVER_assert( __CPROVER_w_ok( pBuffer, bytesToRecv ),
+                      "NetworkInterfaceReceiveStub pBuffer is writable up to bytesToRecv." );
+
+    __CPROVER_havoc_object( pBuffer );
+
+    int32_t bytesOrError;
+    static size_t tries = 0;
+
+    /* It is a bug for the application defined transport send function to return
+     * more than bytesToRecv. */
+    __CPROVER_assume( bytesOrError <= ( int32_t ) bytesToRecv );
+
+    if( tries < ( MAX_NETWORK_RECV_TRIES - 1 ) )
+    {
+        tries++;
+    }
+    else
+    {
+        bytesOrError = 0;
```

Comment:
Should `tries` be reset to `0`?
```suggestion
       tries = 0;    /* Reset the state variable for other calls to the Sntp_ReceiveTimeRespone API. */
        bytesOrError = 0;
```